backend/src/main/java/coe/datacollection/UserService.java

```diff
@@ -148,9 +149,68 @@ public User convertFromDTO(UserDTO dto) {
 		user.setUgMentored(dto.getUgMentored());
 		user.setAwards(dto.getAwards());
 		
-		    user.setTeaching(dto.getTeaching());
-		    user.setClasses(dto.getClasses());
-		    user.setServiceActivity(dto.getServiceActivity());
+		List<Semester> mySemesters = new ArrayList<Semester>();
+		boolean foundSem;
+		
+		for(Teaching i : dto.getTeaching())
+		{
+			i.setUser(user);
+			
+			foundSem = false;
+			for(Semester s : mySemesters)
+			{
+				if(s.getFullName().equals(i.getSemester().getFullName()))
+				{
+					i.setSemester(s);
+					foundSem = true;
+				}
+			}
+			if(!foundSem)
+			{
+				mySemesters.add(i.getSemester());
+			}
+		}
+		user.setTeaching(dto.getTeaching());
+			
+		for(UClasses i : dto.getClasses())
+		{
+			i.setUser(user);
+			
+			foundSem = false;
+			for(Semester s : mySemesters)
+			{
+				if(s.getFullName().equals(i.getSemester().getFullName()))
+				{
+					i.setSemester(s);
+					foundSem = true;
+				}
+			}
+			if(!foundSem)
+			{
+				mySemesters.add(i.getSemester());
+			}
+		}
+		user.setClasses(dto.getClasses());
+			
+		for(UServices i : dto.getServiceActivity())
+		{
+			i.setUser(user);
+			
+			foundSem = false;
+			for(Semester s : mySemesters)
+			{
+				if(s.getFullName().equals(i.getSemester().getFullName()))
+				{
+					i.setSemester(s);
+					foundSem = true;
+				}
+			}
+			if(!foundSem)
+			{
+				mySemesters.add(i.getSemester());
+			}
+		}
+		user.setServiceActivity(dto.getServiceActivity());
```

Comment:
quick fix for entity duplication

will need more work later but keeps functionality for now